esp/scm/ws_config2.ecm

```diff
@@ -0,0 +1,368 @@
+/*##############################################################################
+
+    HPCC SYSTEMS software Copyright (C) 2017 HPCC Systems®.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+
+
+ESPstruct statusMsgType
+{
+    string msgLevel;                // message level ("info", "warning", "error", "fatal")
+    string nodeId;                  // node identifier that generated this status message
+    string attribute("");           // name of node's attribute generating the message ("" for the node itself)
+    string msg;                     // the message
+    ESParray<string, parentId> parentIdList;
+};
+
+ESPstruct statusType
+{
+    bool error(false);    // true if a message exsits in status (below) with a message level of error or higher
+    ESParray<ESPstruct statusMsgType, statusMsg> status;
+};
+
+
+ESPresponse PassFailResponse
+{
+    bool error(false);
+    string msg("");
+};
+
+
+ESPresponse PassFailWithStatusResponse
+{
+    bool   error(false);
```

Comment:
I tried getting inheritance to work and could not get it to build. There was no information in the docs and not much help otherwise. I can revisit if needed. After talking with Tony, I believe inheritance only works for ESPstructs and not ESPrequests or responses. Please let me know if I should try to use inheritance or if it is OK the way it is.

Reply:
agreed.